src/eos-metrics-instrumentation.c

```diff
@@ -351,6 +339,34 @@ record_stop_for_login (GQuark   session_id_quark,
                                           NULL /* auxiliary_payload */);
 }
 
+static void
+add_session (const gchar *session_id,
+             guint32      user_id)
+{
+    if (!add_session_to_set (session_id))
+      return;
+
+    GVariant *user_id_variant = ((user_id >= MIN_HUMAN_USER_ID) ?
+                                 g_variant_new_uint32 (user_id) : NULL);
```

Comment:
Subjective -- I actually prefer my version, but I'll change it since you're the maintainer.